site/_posts/2023-02-10-nextjs-single-sign-on.md

```diff
@@ -0,0 +1,231 @@
+
+{% include posts/sso/register-user.md %}
+
+### Kickstart
+
+Instead of manually setting up FusionAuth using the admin UI as you did above, you can use Kickstart. This tool allows you to get going quickly if you have a fresh installation of FusionAuth. Learn more about how to use [Kickstart](https://fusionauth.io/docs/v1/tech/installation-guide/kickstart).
+
+Here's an example [Kickstart file](https://github.com/FusionAuth/fusionauth-example-nextjs-single-sign-on/blob/main/kickstart/kickstart.json) which sets up FusionAuth for this tutorial.
+
+## Method 1: Clone the demo repository
+
+If you want to run an already working application, you can clone the demo project from this [GitHub repository](https://github.com/FusionAuth/fusionauth-example-nextjs-single-sign-on) using the following command.
+
+```shell
+$ git clone git@github.com:FusionAuth/fusionauth-example-nextjs-single-sign-on.git
+```
+
+Even though you do not need to actually execute the next steps, we do recommend that you read them to understand what was done in the demo application. After that, continue configuring the [Environment variables](#environment-variables).
+
+## Method 2: Create your own Next.js application
+
+If you want to create your own application instead of using our demo project, you can create a new Next.js application by running the command below.
+
+```shell
+$ npx create-next-app name-of-your-application
+```
+
+If it is your first time creating a Next.js application, you'll be prompted to install the `create-next-app` package. Just type `y` to accept it.
+
+You can answer the questions from the wizard any way you want. Here are the ones that we used in our demo project.
+
+```
+✔ Would you like to use TypeScript with this project? … No
+✔ Would you like to use ESLint with this project? … Yes
+✔ Would you like to use `src/` directory with this project? … Yes
+? Would you like to use experimental `app/` directory with this project? … No
+✔ What import alias would you like configured? … @/*
+```
+
+There are several ways of [implementing authentication with Next.js](https://nextjs.org/docs/authentication, and we'll use [NextAuth.js](https://next-auth.js.org), a complete open-source authentication solution, in this article.
+
+### Installing NextAuth.js
+
+{% include _callout-note.liquid content="We recommend you look at the [NextAuth.js Getting Start guide](https://next-auth.js.org/getting-started/example) for more up-to-date instructions." %}
+
+First, install NextAuth.js via one of the following three methods.
+
+```shell
+# Using npm
+$ npm install next-auth
+
+# Using yarn
+$ yarn add next-auth
+
+# Using pnpm
+$ pnpm add next-auth
+```
+
+Now, you need to create a file named exactly like `[...nextauth].js` in `src/pages/api/auth` (you need to create the `auth` folder there) and configure the [built-in support for FusionAuth](https://next-auth.js.org/providers/fusionauth). This will make every request to `/api/auth/*` handled by NextAuth.js.
```

Comment:
```suggestion
Now, you need to create a file named exactly like `[...nextauth].js` in `src/pages/api/auth`.

First, make the directory:

``shell
mkdir src/pages/api/auth 
``

Then create a file named `[...nextauth].js` in that directory.

Next, you'll configure the [built-in support for FusionAuth](https://next-auth.js.org/providers/fusionauth). Doing so ensures every request to the `/api/auth/*` path is handled by NextAuth.js.
```

Makes sure to add the extra backtick to the mkdir section. I couldn't have it have 3 backticks without ruining the suggestion.